src/sys/mod.rs

```diff
@@ -160,6 +160,23 @@ impl fmt::Display for HandleKind {
     }
 }
 
+/// Reference point and offset to seek a position in a stream with
+pub enum StreamSeek {
+    /// Seek a position forward from the start of the stream
+    Start(usize),
+    /// Seek a position backward from the end of the stream
+    End(usize),
+}
+
+impl Into<std::io::SeekFrom> for StreamSeek {
+    fn into(self) -> std::io::SeekFrom {
+        match self {
+            StreamSeek::Start(off) => std::io::SeekFrom::Start(off as u64),
+            StreamSeek::End(off) => std::io::SeekFrom::End(-(off as i64)),
+        }
+    }
+}
```

Comment:
```suggestion
impl From<StreamSeek> for std::io::SeekFrom {
    fn from(value: StreamSeek) -> Self {
        match value {
            StreamSeek::Start(off) => std::io::SeekFrom::Start(off as u64),
            StreamSeek::End(off) => std::io::SeekFrom::End(-(off as i64)),
        }
    }
}
```
Clippy says to implement `From` instead of `Into` because the former gives you the latter but not the other way around.